File: app/src/main/java/ru/art2000/calculator/view/unit/AllUnitsActivity.java
package ru.art2000.calculator.view.unit;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.core.app.ActivityCompat;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.Objects;

import ru.art2000.calculator.R;
import ru.art2000.calculator.databinding.ActivityAllUnitsBinding;
import ru.art2000.calculator.view_model.currency.CurrenciesSettingsModel;
import ru.art2000.calculator.view_model.unit.UnitConverterDependencies;
import ru.art2000.calculator.view_model.unit.UnitConverterModel;
import ru.art2000.extensions.activities.AutoThemeActivity;

public class AllUnitsActivity extends AutoThemeActivity {



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityAllUnitsBinding binding = ActivityAllUnitsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        UnitConverterModel model = new ViewModelProvider(this).get(UnitConverterModel.class);

        String category = getIntent().getStringExtra("category");

        setSupportActionBar(binding.toolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        String title = getResources().getStringArray(R.array.unit_converter_categories)
                [UnitConverterDependencies.getCategoryInt(category)];
        getSupportActionBar().setTitle(title);

        int pos = getIntent().getIntExtra("highlightPosition", 0);

        RecyclerView list = binding.allUnitsList;
        list.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
        LinearLayoutManager llm = new LinearLayoutManager(this);
        llm.setOrientation(RecyclerView.VERTICAL);
        list.setLayoutManager(llm);
        UnitListAdapter adapter = new UnitListAdapter(this, this,
                UnitConverterDependencies.getCategoryItems(category), model, pos);
        list.setAdapter(adapter);
    }

    @Override
    public boolean onSupportNavigateUp() {
        finish();
        return true;
    }
}

File: app/src/main/java/ru/art2000/calculator/view/unit/AllUnitsActivity.kt
package ru.art2000.calculator.view.unit

import android.os.Bundle
import android.view.View
import by.kirich1409.viewbindingdelegate.CreateMethod
import by.kirich1409.viewbindingdelegate.viewBinding
import dagger.hilt.android.AndroidEntryPoint
import ru.art2000.calculator.R
import ru.art2000.calculator.databinding.ActivityAllUnitsBinding
import ru.art2000.calculator.model.unit.UnitCategory
import ru.art2000.calculator.view.AppActivity
import ru.art2000.calculator.view_model.unit.UnitConverterModel
import ru.art2000.extensions.activities.getEnumExtra
import ru.art2000.extensions.arch.assistedViewModel
import ru.art2000.extensions.views.OrientationManger
import ru.art2000.extensions.views.addOrientationItemDecoration
import javax.inject.Inject

@AndroidEntryPoint
class AllUnitsActivity : AppActivity() {

    @Inject
    lateinit var viewModelFactory: UnitConverterModel.Factory

    private val category by lazy {
        intent.getEnumExtra<UnitCategory>("category")!!
    }

    private val model by assistedViewModel { viewModelFactory.create(category) }
    private val binding by viewBinding<ActivityAllUnitsBinding>(CreateMethod.INFLATE)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(binding.root)

        val pos = intent.getIntExtra("highlightPosition", 0)

        setSupportActionBar(binding.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        val title =
            resources.getStringArray(R.array.unit_converter_categories)[category.ordinal]
        supportActionBar?.title = title

        val unitListAdapter = UnitListAdapter(
            this, this,
            model.converterFunctions, model::copy, pos,
        )

        binding.allUnitsList.apply {
            layoutManager = OrientationManger(this@AllUnitsActivity)
            addOrientationItemDecoration()
            adapter = unitListAdapter
        }
    }

    override val topViews: List<View>
        get() = listOf(binding.root)

    override val bottomViews: List<View>
        get() = listOf(binding.allUnitsList)

    override val leftViews: List<View>
        get() = listOf(binding.root)

    override val rightViews: List<View>
        get() = listOf(binding.root)

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}